V1 Done and working

Trubador opened the pull request.

Trubador committed:
V1 done

Trubador committed:
added http action task block

Trubador committed:
Added additional Action Task blocks and updated sample flow

Trubador committed:
Lots of improvements

Trubador committed:
Merge branch 'DidactHQ:main' into main

Trubador committed:
Create main.yml

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/DidactHQ/didact-core?pullRequest=5) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/DidactHQ/didact-core?pullRequest=5) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/DidactHQ/didact-core?pullRequest=5) it.</sub>

Trubador committed:
Add ExecutionMode support and refactor database initialization

- Introduced ExecutionMode entity and updated related configurations.
- Added FlowRunRepository for managing flow runs.
- Updated database context to include ExecutionModes DbSet.
- Refactored database initialization logic to seed ExecutionModes.
- Updated application to use SQLite as the default database provider.
- Enhanced error handling and logging in various components.
